taskcluster/scripts/setup-mac-worker.sh

```diff
@@ -7,6 +7,8 @@
 mkdir -p "$HOME/bin"
 export PATH="$HOME/bin:$HOME/Library/Python/3.7/bin:$PATH"
 
+pip3 install pytest
```

Comment:
It's probably worth pinning this, otherwise pytest is going to make some backwards incompatible change at some point that breaks CI outside of a commit.

It might also be worth using a `requirements.txt` generated via `pip-compile` that includes hashes, but it's just for testing so up to you.

Reply:
I'm not going to do this in this PR; we're using `pip install` and `gem`  in several places without hashes. This is a task on its own.

Reply:
Sounds good, fwiw this can be as simple as changing this to `pip3 install pytest~=7.4`. But keeping scope down is reasonable.